resources/todomvc/vanilla-examples/javascript-es6/dist/template.js

```diff
@@ -47,26 +49,24 @@ class Template {
      * })
      */
     show(data) {
-        let i = 0;
         let view = "";
-        const len = data.length;
 
-        for (i; i < len; i++) {
-            let completed = "";
-            let checked = "";
-            let template = this.defaultTemplate;
+        for (let i = 0; i < data.length; i++) {
+            const completed = data[i].completed ? "completed" : "";
+            const checked = data[i].completed ? "checked" : "";
 
-            if (data[i].completed) {
-                completed = "completed";
-                checked = "checked";
-            }
+            const valuesToReplace = {
+                "{{id}}": data[i].id,
+                "{{title}}": escape(data[i].title),
+                "{{completed}}": completed,
+                "{{checked}}": checked,
+                "{{list-index}}": i,
+                "{{view-index}}": i,
+            };
 
-            template = template.replace("{{id}}", data[i].id);
-            template = template.replace("{{title}}", escape(data[i].title));
-            template = template.replace("{{completed}}", completed);
-            template = template.replace("{{checked}}", checked);
-            template = template.replace("{{list-index}}", i);
-            template = template.replace("{{view-index}}", i);
```

Comment:
@rniwa this is mostly to remove this, given that this makes `String.prototype.replace` very prominent for the 'add todo items' step.